src/lib/message/L1Transaction.ts

```diff
@@ -103,28 +104,35 @@ export class L1TransactionReceipt implements TransactionReceipt {
    * Get any MessageDelivered events that were emitted during this transaction
    * @returns
    */
-  public getMessageDeliveredEvents(): MessageDeliveredEvent['args'][] {
+  public getMessageDeliveredEvents(): EventArgs<MessageDeliveredEvent>[] {
     const iface = Bridge__factory.createInterface()
     const messageDeliveredTopic = iface.getEventTopic(
       iface.getEvent('MessageDelivered')
     )
     return this.logs
       .filter(log => log.topics[0] === messageDeliveredTopic)
-      .map(l => iface.parseLog(l).args as MessageDeliveredEvent['args'])
+      .map(
+        l =>
+          iface.parseLog(l).args as unknown as EventArgs<MessageDeliveredEvent>
```

Comment:
are you thinking of a type guard that just throws if the type is invalid? That would add a slight bit of protection, but only one the fields tested. So to have full coverage we'd have to type guard test all the event fields, which may be a bit overkill.

Adding a type guard would be nice, as it would improve error messaging, but if you're program relies on one of those fields in the event and it isnt there then the program will fail either way (probably an object undefined except later), but with a type guard it's more graceful

Reply:
Mmm, actually, the `parseLog` should fail, since the topic would be different if we change the event. So event is guaranteed to exist. I dont think a type guard adds anything.